File: GestaoLivrosApi/GestaoLivrosApi/Models/Authors/EditAuthorDTO.cs
using System;
using FluentValidation;

namespace GestaoLivrosApi.Models.Authors
{
	public class EditAuthorDTO
	{
        public int Id { get; set; }
        public string Name { get; set; } = null!;

        public string Country { get; set; } = null!;

        public string? Image { get; set; }

        public class EditAuthorDTOValidator : AbstractValidator<EditAuthorDTO>
        {
            public EditAuthorDTOValidator()
            {
                RuleFor(a => a.Name).NotNull().WithMessage("Insira o Nome do autor");
                RuleFor(a => a.Country).NotNull().WithMessage("Insira o País do autor");
            }
        }
    }
}

File: frontend/src/models/Authors/EditAuthorDTO.ts
import Joi from "joi";

export class EditAuthorDTO{
    id:number = 0;
    name:string = "";
    country:string = "";
    image:string = "";
    books:[] = [];
}

export const EditAuthorDTOSchema = Joi.object({
    name: Joi.string().messages({"string.empty":"Nome do autor deve ser preenchido"}),
    country: Joi.string().messages({"string.empty":"País do autor deve ser preenchido"})
})